flytepropeller/pkg/controller/config/config.go

```diff
@@ -259,6 +259,7 @@ const (
 type EventConfig struct {
 	RawOutputPolicy           RawOutputPolicy `json:"raw-output-policy" pflag:",How output data should be passed along in execution events."`
 	FallbackToOutputReference bool            `json:"fallback-to-output-reference" pflag:",Whether output data should be sent by reference when it is too large to be sent inline in execution events."`
+	ErrorOnAlreadyExists      bool            `json:"error-on-already-exists" pflag:",Whether to return an error when an event already exists."`
```

Comment:
worth enabling this [in single-binary](https://github.com/flyteorg/flyte/blob/ff12f36d385cdcc0ad2acc86803940cd3812ea23/charts/flyte-binary/templates/configmap.yaml#L42-L43) by default?

Reply:
This should never really be enabled except for array node. I didn't want to update the record task event function, but I guess that might be better if users were to unknowingly change this and have executions fail.

Reply:
actually I think I can do this a little better and not expose a new config